hankel/hankel.py

```diff
@@ -179,40 +125,69 @@ def transform(self, f, k=1, ret_err=True, ret_cumsum=False, inverse=False):
         ret : array-like
             The Hankel-transform of f(x) at the provided k. If
             `k` is scalar, then this will be scalar.
-
         err : array-like
             The estimated error of the approximate integral, at every `k`.
             It is merely the last term in the sum.
             Only returned if `ret_err=True`.
-
         cumsum : array-like
             The total cumulative sum, for which the last term
             is itself the transform.
             One can use this to check whether the integral is converging.
             Only returned if `ret_cumsum=True`
 
-
         Notes
         -----
         The Hankel transform is defined as
 
         .. math:: F(k) = \int_0^\infty r f(r) J_\nu(kr) dr.
 
+        Or in the alternative case with ``alt=True``:
+
+        .. math:: F(k) = \int_0^\infty f(r) \sqrt{kr} J_\nu(kr) dr.
+
         The inverse transform is identical (swapping *k* and *r* of course).
         """
         # The following allows for a re-scaling of k when doing FT's.
+        k_scalar = np.isscalar(k)
         k = self._k(k)
+        # k = zero here
+        k_0 = np.isclose(k, 0)
 
         # The following is the scalar normalisation of the transform
         # The basic transform has a norm of 1.
         # But when doing FT's, this depends on the dimensionality.
         norm = self._norm(inverse)
 
         # The following renormalises by the fourier dual to some power
-        knorm = k ** self._k_power
+        knorm = safe_power(k, self._k_power)
+
+        # set replace k=0 with 1 in the following
+        knorm[k_0] = 1
+        k[k_0] = 1
 
         summation = self._get_series(f, k)
-        ret = norm * np.sum(summation, axis=-1) / knorm
+        ret = np.array(norm * np.sum(summation, axis=-1) / knorm)
+
+        # care about k = 0
+        if np.any(k_0):
+            # limit of J(nu, 0) considering powers of k
+            lim_k_pow = 0.5 if self.alt else 0  # in alt. def sqrt(k) involved
+            lim_r_pow = self._r_power + lim_k_pow + self.nu
+            nu_th = self._k_power - 1 - lim_k_pow  # threshold
```

Comment:
what does `nu_th` represent?

Reply:
This is the threshold for nu determining the behavior of the integral for k -> 0.